xskillscore/core/probabilistic.py

```diff
@@ -526,73 +622,153 @@ def rps(
     if fair:
-        e = forecasts
+        M = forecasts[member_dim].size
 
-    # normalize f.sum()=1
-    forecasts = forecasts / forecasts.sum(bin_dim)
-    observations = observations / observations.sum(bin_dim)
+    forecasts = _bool_to_int(forecasts)
 
-    Fc = forecasts.cumsum(bin_dim)
-    Oc = observations.cumsum(bin_dim)
+    _check_identical_xr_types(observations, forecasts)
 
-    if fair:
-        Ec = e.cumsum(bin_dim)
-        res = (((Ec / M) - Oc) ** 2 - Ec * (M - Ec) / (M ** 2 * (M - 1))).sum(bin_dim)
+    # different entry point of calculating RPS based on category_edges
+    if isinstance(category_edges, np.ndarray):
+        # prepare category_edges as xr object
+        category_edges = xr.DataArray(
+            category_edges,
+            dims="category_edge",
+            coords={"category_edge": category_edges},
+        )
+        category_edges = xr.ones_like(observations) * category_edges
+
+    if isinstance(category_edges, (xr.Dataset, xr.DataArray)) or isinstance(
+        category_edges, tuple
+    ):
+        if isinstance(
+            category_edges, tuple
+        ):  # edges tuple of two: use for obs and forecast edges separately
+            observations_edges, forecasts_edges = category_edges
+            _check_identical_xr_types(forecasts_edges, forecasts)
+            _check_identical_xr_types(observations_edges, forecasts)
+        else:  # edges only given once, so use for both obs and forecasts
+            _check_identical_xr_types(category_edges, forecasts)
+            observations_edges, forecasts_edges = category_edges, category_edges
+
+        _check_data_within_edges(forecasts, forecasts_edges)
+        _check_data_within_edges(observations, observations_edges)
+
+        forecasts_edges = _add_eps_to_last_in_dim(forecasts_edges, bin_dim)
+        observations_edges = _add_eps_to_last_in_dim(observations_edges, bin_dim)
+
+        # cumulative probs, ignore lowest threshold as below category_edges
+        Fc = (
+            (forecasts < forecasts_edges)
+            .mean(member_dim)
+            .isel({bin_dim: slice(1, None)})
+        )
+        Oc = (
+            (observations < observations_edges)
+            .astype("int")
+            .isel({bin_dim: slice(1, None)})
+        )
+
+    elif category_edges is None:  # expect CDFs already as inputs
+        if member_dim in forecasts.dims:
+            forecasts = forecasts.mean(member_dim)
+        Fc = forecasts
+        Oc = observations
     else:
+        raise ValueError(
+            f"category_edges must be xr.DataArray, xr.Dataset, tuple of xr.objects, None or array-like, found {type(category_edges)}"
+        )
+
+    _check_is_CDF(Fc)
+    _check_is_CDF(Oc)
+
+    # RPS formulas
+    if fair:  # for ensemble member adjustment Ferro 2013
+        Ec = Fc * M
+        res = ((Ec / M - Oc) ** 2 - Ec * (M - Ec) / (M ** 2 * (M - 1))).sum(bin_dim)
+    else:  # normal formula
         res = ((Fc - Oc) ** 2).sum(bin_dim)
```

Comment:
@judithberner I had already implemented the ensemble member adjustment based on Ferro 2013 C. A. T. Ferro. Fair scores for ensemble forecasts. Q.R.J. Meteorol. Soc., 140: 1917–1923, 2013. doi: 10.1002/qj.2270.